wrappers/golang/core/slice.go

```diff
@@ -54,6 +54,65 @@ func (d DeviceSlice) CheckDevice() {
 	}
 }
 
+func (d *DeviceSlice) Range(start, end int, endInclusive bool) DeviceSlice {
+	if end <= start {
+		panic("Cannot have negative or zero size slices")
+	}
+
+	if end >= d.length {
+		panic("Cannot increase slice size from Range")
+	}
+
+	var newSlice DeviceSlice
+	switch {
+	case start < 0:
+		panic("Negative value for start is not supported")
+	case start == 0:
+		newSlice = d.RangeTo(end, endInclusive)
+	case start > 0:
+		tempSlice := d.RangeFrom(start)
+		newSlice = tempSlice.RangeTo(end-start, endInclusive)
+	}
+	return newSlice
+}
+
+func (d *DeviceSlice) RangeTo(end int, inclusive bool) DeviceSlice {
+	if end <= 0 {
+		panic("Cannot have negative or zero size slices")
+	}
+
+	if end >= d.length {
+		panic("Cannot increase slice size from Range")
+	}
+
+	var newSlice DeviceSlice
+	sizeOfElement := d.capacity / d.length
+	newSlice.length = end
+	if inclusive {
+		newSlice.length += 1
+	}
+	newSlice.capacity = newSlice.length * sizeOfElement
+	newSlice.inner = d.inner
+	return newSlice
+}
+
+func (d *DeviceSlice) RangeFrom(start int) DeviceSlice {
+	if start >= d.length {
+		panic("Cannot have negative or zero size slices")
+	}
+
```

Comment:
A check should be added here
```
if start < 0 {
    panic("Negative value for start is not supported")
}
```
Also checks inside `Range` from `RangeFrom` and `RangeTo` can be removed optionally